presto-main/src/main/java/com/facebook/presto/type/DecimalOperators.java

```diff
@@ -616,4 +619,28 @@ public static long hashCode(@SqlType("decimal(p, s)") Slice value)
             return XxHash64.hash(value);
         }
     }
+
+    @ScalarOperator(INDETERMINATE)
+    public static final class Indeterminate
+    {
+        @LiteralParameters({"p", "s"})
+        @SqlType(StandardTypes.BOOLEAN)
+        public static boolean indeterminate(@SqlType("decimal(p, s)") long value, @IsNull boolean isNull)
+        {
+            if (isNull) {
+                return true;
+            }
+            return false;
+        }
+
+        @LiteralParameters({"p", "s"})
+        @SqlType(StandardTypes.BOOLEAN)
+        public static boolean indeterminate(@SqlType("decimal(p, s)") Slice value, @IsNull boolean isNull)
```

Comment:
I don't think the `@IsNull` argument is needed here. You can mark the `value` parameter as `@Nullable` and check it if's null directly -- it's already boxed.